emacs/tern.el

```diff
@@ -293,15 +293,17 @@ list of strings, giving the binary name and arguments.")
     (when (and opening-paren (equal (char-after opening-paren) ?\())
       (if (and tern-last-argument-hints (eq (car tern-last-argument-hints) opening-paren))
           (tern-show-argument-hints)
-        (tern-run-query (lambda (data)
-                          (let ((type (tern-parse-function-type data)))
-                            (when type
-                              (setf tern-last-argument-hints (cons opening-paren type))
-                              (tern-show-argument-hints))))
-                        `((type . "type")
-                          (preferFunction . t))
-                        opening-paren
-                        :silent)))))
+        (lexical-let ((generation tern-command-generation))
+          (tern-run-query (lambda (data)
+                            (when (= tern-command-generation (1- generation))
```

Comment:
My memory fails me on this. 
I remember I was being aggressive in denying previous requests from executing.
Testing out the code again, you might be right.